controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -332,12 +332,15 @@ public Void initializeStreamWriters(final String commitStreamName, final EventSt
                 log.debug("Txn={}, notified segments stores", txnId));
 
         // Step 5. Start tracking txn in timeout service
-        return notify.thenApplyAsync(y -> {
-            int version = txnFuture.join().getVersion();
-            long executionExpiryTime = txnFuture.join().getMaxExecutionExpiryTime();
+        return notify.whenCompleteAsync((result, ex) -> {
+            int version = 0;
+            long executionExpiryTime = System.currentTimeMillis() + maxExecutionPeriod;
```

Comment:
In case of failure creating a transaction, we would like to compensate for all the previous activities. `abortTransaction` method would do that, but that method could fail as well, and would require multiple retries for successful execution. Hence, in this PR, we update in-memory structures to record cleaning up of partially created transaction - an operation that cannot fail. Consequently, transaction would be cleaned up in the background with appropriate retries.